imas/ids_convert.py

```diff
@@ -919,3 +944,82 @@ def _ids_properties_source(source: IDSString0D, provenance: IDSStructure) -> Non
     provenance.node.resize(1)
     provenance.node[0].reference.resize(1)
     provenance.node[0].reference[0].name = source.value
+
+
+def _pulse_schedule_3to4(
```

Comment:
should these go in a separate file perhaps? or do we do that when we add another large migration?

Reply:
We could, but these `_pulse_schedule_3to4*` functions also depend on some of the other functions and globals from this module (e.g. `_get_target_item`, and `logger`).

Separating them into separate files is slightly messy with circular imports etc. At this point that is not really worth it IMO :)